pyro/infer/mcmc/util.py

```diff
@@ -0,0 +1,105 @@
+from collections import defaultdict
+
+from pyro.distributions.util import log_sum_exp
+from pyro.infer.util import is_validation_enabled
+from pyro.util import check_site_shape
+
+
+class EnumTraceProbEvaluator(object):
+    def __init__(self,
+                 model_trace,
+                 has_enumerable_sites=False,
+                 max_iarange_nesting=float("inf")):
+        """
```

Comment:
nit: could you move this comment to a class-level comment rather than an `__init__` comment?

Reply:
My bad, will change.